Enable/Disable All Addons

RedGuy12 opened the pull request:
**Resolves**

Resolves #318 
Resolves #1211

**Changes**

added enabling/disabling all addons
added expand/collapse all addons

**Reason for changes**

people want them

**Tests**

<!-- Have you tested this pull request? If so, how? -->
On Microsoft Edge Version 87.0.664.66 (Official Build) (64-bit)

Needs Firefox tests

RedGuy12 commented:
Wait #1200 #1200 #1200 #1200 #1200 Yayyyy

RedGuy12 commented:
How do I run Prettier?

apple502j commented:
It doesn't seem like this PR is tested at all. Closing for now.

RedGuy12 commented:
I did test it...
________________________________
From: apple502j <notifications@github.com>
Sent: Monday, January 4, 2021 12:05:04 PM
To: ScratchAddons/ScratchAddons <ScratchAddons@noreply.github.com>
Cc: Paul Reid <paul@reid-family.org>; Author <author@noreply.github.com>
Subject: Re: [ScratchAddons/ScratchAddons] Enable/Disable All Addons (#1200)


It doesn't seem like this PR is tested at all. Closing for now.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ScratchAddons/ScratchAddons/pull/1200#issuecomment-754126687>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOT5DEXM4OULLSTVTPQTQUDSYH7NBANCNFSM4VTI4KPQ>.

RedGuy12 commented:
Oh wait I never looked at light mode
________________________________
From: Paul Reid <paul@reid-family.org>
Sent: Monday, January 4, 2021 12:26:46 PM
To: ScratchAddons/ScratchAddons <reply@reply.github.com>; ScratchAddons/ScratchAddons <ScratchAddons@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: Re: [ScratchAddons/ScratchAddons] Enable/Disable All Addons (#1200)

I did test it...
________________________________
From: apple502j <notifications@github.com>
Sent: Monday, January 4, 2021 12:05:04 PM
To: ScratchAddons/ScratchAddons <ScratchAddons@noreply.github.com>
Cc: Paul Reid <paul@reid-family.org>; Author <author@noreply.github.com>
Subject: Re: [ScratchAddons/ScratchAddons] Enable/Disable All Addons (#1200)


It doesn't seem like this PR is tested at all. Closing for now.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ScratchAddons/ScratchAddons/pull/1200#issuecomment-754126687>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOT5DEXM4OULLSTVTPQTQUDSYH7NBANCNFSM4VTI4KPQ>.

RedGuy12 commented:
reopen this?

RedGuy12 commented:
repoen this pls? i can fix the light mode bugs

RedGuy12 commented:
Please reopen this, I can fix it. If you don't what to, please tell me _why_.

RedGuy12 commented:
I fixed it!!

WorldLanguages commented:
This will likely conflict with #1006 

Moving to v1.8 milestone

GrahamSH-LLK commented on webpages/settings/index.js:
```diff
@@ -650,6 +650,41 @@ document.addEventListener("keydown", (e) => {
 
 chrome.runtime.sendMessage("checkPermissions");
 
+document.getElementById("enableAll").addEventListener("click", () => {
+  Object.values(
+    document.querySelectorAll('.addon-body:not([style="display: none;"]) .switch[state=off]')
+  ).forEach((s) => s.click());
+});
+
+document.getElementById("disableAll").addEventListener("click", () => {
+  Object.values(document.querySelectorAll('.addon-body:not([style="display: none;"]) .switch[state=on]')).forEach((s) =>
+    s.click()
+  );
+});
+
```
Use Vue @click, not event listeners for all of these, not just the ones I have selected

RedGuy12 commented:
Ok I dont know vue but I'll figure it out
________________________________
From: GrahamSH <notifications@github.com>
Sent: Saturday, January 9, 2021 9:05:25 AM
To: ScratchAddons/ScratchAddons <ScratchAddons@noreply.github.com>
Cc: Paul Reid <paul@reid-family.org>; Author <author@noreply.github.com>
Subject: Re: [ScratchAddons/ScratchAddons] Enable/Disable All Addons (#1200)


@GrahamSH-LLK commented on this pull request.

________________________________

In webpages/settings/index.js<https://github.com/ScratchAddons/ScratchAddons/pull/1200#discussion_r554436741>:

> +document.getElementById("enableAll").addEventListener("click", () => {
+  Object.values(
+    document.querySelectorAll('.addon-body:not([style="display: none;"]) .switch[state=off]')
+  ).forEach((s) => s.click());
+});
+
+document.getElementById("disableAll").addEventListener("click", () => {
+  Object.values(document.querySelectorAll('.addon-body:not([style="display: none;"]) .switch[state=on]')).forEach((s) =>
+    s.click()
+  );
+});
+


Use Vue @click<https://github.com/click>, not event listeners for all of these, not just the ones I have selected

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ScratchAddons/ScratchAddons/pull/1200#pullrequestreview-564789206>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOT5DEVJOEJST6RIDSHGZHDSZBWDLANCNFSM4VTI4KPQ>.

WorldLanguages commented:
This works, but is not executed very well.
1. Ideally we'd directly use Vue data and methods to enable addons instead of triggering events on the DOM
2. These buttons should not appear in "popup mode" (addons tab in the extension popup)
3. "Enable all" is just a terrible idea - some addons require permission prompts, some are beta, some are very preference based. At most, "enable all recommended addons" would be fine, if the UI/UX is done right.
4. I don't see any use for "disable addons", I'd rather see users disable the whole extension so that they can keep their settings.
5. "Invert selected addons" is just bizarre
6. Expand/collapse all are fine, but I'd prefer a single button, to the left of the search bar. By default enabled addons are expanded, so this button should say "collapse all" unless all addons are collapsed (a computed property could be used to check the former)

RedGuy12 commented:
> * Ideally we'd directly use Vue data and methods to enable addons instead of triggering events on the DOM

done, needs to be uploaded

> * These buttons should not appear in "popup mode" (addons tab in the extension popup)

done, needs to be uploaded

> * "Enable all" is just a terrible idea - some addons require permission prompts, some are beta, some are very preference based. At most, "enable all recommended addons" would be fine, if the UI/UX is done right.

uhhhhh see https://github.com/ScratchAddons/ScratchAddons/issues/318#issue-705827136 that's the main purpose of the issue...
plus I only have like 2 not enabled

> * I don't see any use for "disable addons", I'd rather see users disable the whole extension so that they can keep their settings.

bug testing - someone has a ton of addons enabled, but doesn't want any enabled for a minute to see which addons cause a certain bug

> * "Invert selected addons" is just bizarre

ok

> * Expand/collapse all are fine, but I'd prefer a single button, to the left of the search bar. By default enabled addons are expanded, so this button should say "collapse all" unless all addons are collapsed (a computed property could be used to check the former)

ok

WorldLanguages commented:
> uhhhhh see #318 (comment) that's the main purpose of the issue...
plus I only have like 2 not enabled

I wrote on https://github.com/ScratchAddons/ScratchAddons/issues/318#issuecomment-729225302 that it needed further discussion. I added it to the backlog because I considered it something to explore, not because the implementation in the first comment of the issue was ideal.
You're trying to tackle too many issues with this PR (mass enable functionality, its UI and collapse/expand all) when the first 2 haven't been discussed enough. Doing this with addons is usually fine (since 90% of addons are disabled by default), but changing the settings page affects all users.
Can you first try to work on the former first? I detailed how it should work in the previous comment (see 6)

RedGuy12 commented:
Ok, I moved it to @click already and removed it from the popup. Lemme work on the others
________________________________
From: World_Languages <notifications@github.com>
Sent: Friday, January 15, 2021 1:12:21 PM
To: ScratchAddons/ScratchAddons <ScratchAddons@noreply.github.com>
Cc: Paul Reid <paul@reid-family.org>; Comment <comment@noreply.github.com>
Subject: Re: [ScratchAddons/ScratchAddons] Enable/Disable All Addons (#1200)


uhhhhh see #318<https://github.com/ScratchAddons/ScratchAddons/issues/318> (comment) that's the main purpose of the issue...
plus I only have like 2 not enabled

I wrote on #318 (comment)<https://github.com/ScratchAddons/ScratchAddons/issues/318#issuecomment-729225302> that it needed further discussion. I added it to the backlog because I considered it something to explore, not because the implementation in the first comment of the issue was ideal.
You're trying to tackle too many issues with this PR (mass enable functionality, its UI and collapse/expand all) when the first 2 haven't been discussed enough. Doing this with addons is usually fine (since 90% of addons are disabled by default), but changing the settings page affects all users.
Can you first try to work on the former first? I detailed how it should work in the previous comment (see 6)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/ScratchAddons/ScratchAddons/pull/1200#issuecomment-761132953>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOT5DEXVOOK7O3GRPV7ZFW3S2CHRLANCNFSM4VTI4KPQ>.

RedGuy12 commented:
will open new pr when i have time
